server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+        ret = repositories._merge_related_objects('importers', mock_importer_manager, mock_repos)
+
+        self.assertTrue(len(ret) == 2)
+        self.assertEqual(map(itemgetter('id'), ret), ['mock1', 'mock2'])
+        mock1_expected_importers = []
+        mock2_expected_importers = []
+        mock1_importers = ret[map(itemgetter('id'), ret).index('mock1')]['importers']
+        mock2_importers = ret[map(itemgetter('id'), ret).index('mock2')]['importers']
+        self.assertEqual(mock1_importers, mock1_expected_importers)
+        self.assertEqual(mock2_importers, mock2_expected_importers)
+
+
+class TestConvertRepoDatesToStrings(unittest.TestCase):
+    """
+    Tests the standardization of dates in repo fields.
+    """
+
+    def test_last_unit_added_and_removed(self):
+        """
+        Test that last_unit_added and last_unit_removed fields contain properly formatted dates.
+        """
+
+        dt = datetime.datetime.utcnow()
+        repo = {'id': 'dummy-1', 'display_name': 'dummy',
+                'last_unit_added': dt,
+                'last_unit_removed': dt}
+        string_date = dateutils.format_iso8601_datetime(
+            dateutils.to_utc_datetime(dt, no_tz_equals_local_tz=False))
+        repositories._convert_repo_dates_to_strings(repo)
+        self.assertEquals(repo['last_unit_added'], string_date)
+        self.assertEquals(repo['last_unit_removed'], string_date)
+
+    @mock.patch('pulp.server.webservices.views.repositories.dateutils.to_utc_datetime')
+    @mock.patch('pulp.server.webservices.views.repositories.dateutils.format_iso8601_datetime')
+    def test_no_last_added_or_removed(self, mock_to_utc, mock_8601):
+        """
+        When last_unit_added or last_unit_removed do not exist, they should be ignored.
+        """
+
+        repo = {'id': 'dummy-1', 'display_name': 'dummy',
+                'last_unit_added': None,
+                'last_unit_removed': None}
+        repositories._convert_repo_dates_to_strings(repo)
+        self.assertEquals(repo['last_unit_added'], None)
+        self.assertEquals(repo['last_unit_removed'], None)
+        self.assertFalse(mock_to_utc.called)
+        self.assertFalse(mock_8601.called)
+
+
+class TestReposView(unittest.TestCase):
+    """
+    Tests for ReposView.
+    """
+
+    @mock.patch('pulp.server.webservices.views.repositories._convert_repo_dates_to_strings')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    def test__process_repos_minimal(self, mock_reverse, mock_convert_date):
+        """
+        Test _process_repos without optional args, assert that processing was called for each repo.
+        """
+
+        mock_reverse.return_value = '/mock/path/'
+        mock_repos = [{'id': 'mock1'}, {'id': 'mock2'}]
+        repos_view = ReposView()
+        repos_view._process_repos(mock_repos, False, False)
+        mock_convert_date.assert_has_calls(
+            [mock.call({'id': 'mock1', '_href': '/mock/path/'}),
```

Comment:
This is related to how pure of a unit testing methodology we want to have. Pure unit tests don't rely on third party units for their correctness to happen. As such, a unit under test should be fully mocked so that it doesn't make calls outside of the unit under test. Note that unit tests cannot guarantee overall correctness so that is OK. It only asserts correctness on the component under test.

Not mocking `reverse` would be more of an integration test not a unit test because it would test the integration between two components (the unit under test and the reverse component). I advise a pure unit test methodology and also having completely separate set of integration tests. We have those integration tests separate so I think these should be pure unit tests.

I think `reverse` should be mocked because then we can...
* assert that it is called
* assert that we are passing it the right parameters
* assert that the result is returned as _href. Currently `reverse.return_value` returns a string but it could be better to let it return a mock() object as mocks normally do and then assert that the _href response `is reverse.return_value`.